src/units.h

```diff
@@ -369,6 +369,84 @@ inline constexpr double to_kilogram( const mass &v )
     return v.value() / 1000000.0;
 }
 
+
+// Specific energy
+const specific_energy specific_energy_min = units::specific_energy(
+            std::numeric_limits<units::specific_energy::value_type>::min(), units::specific_energy::unit_type{} );
+
+const specific_energy specific_energy_max = units::specific_energy(
+            std::numeric_limits<units::specific_energy::value_type>::max(), units::specific_energy::unit_type{} );
+
+template<typename value_type>
+inline constexpr quantity<value_type, specific_energy_in_joule_per_gram_tag>
+from_joule_per_gram( const value_type v )
+{
+    return quantity<value_type, specific_energy_in_joule_per_gram_tag>( v,
+            specific_energy_in_joule_per_gram_tag{} );
+}
+
+template<typename value_type>
+inline constexpr value_type to_joule_per_gram( const
+        quantity<value_type, specific_energy_in_joule_per_gram_tag> &v )
+{
+    return v.value();
+}
+
+// Temperature
+// Absolute zero - possibly should just be INT_MIN
+const temperature temperature_min = units::temperature( 0, units::temperature::unit_type{} );
+
+const temperature temperature_max = units::temperature(
+                                        std::numeric_limits<units::temperature::value_type>::max(), units::temperature::unit_type{} );
+
+template<typename value_type>
+inline constexpr quantity<value_type, temperature_in_kelvin_tag> from_kelvin(
+    const value_type v )
+{
+    return quantity<value_type, temperature_in_kelvin_tag>( v, temperature_in_kelvin_tag{} );
+}
+
+template<typename value_type>
+inline constexpr value_type to_kelvin( const
+        quantity<value_type, temperature_in_kelvin_tag> &v )
+{
+    return v.value();
+}
+
+constexpr temperature freezing_point = from_kelvin( 273.150f );
+//constexpr temperature freezing_point = from_kelvin( 459.67f );
+
+template<typename value_type>
+inline constexpr quantity<value_type, temperature_in_kelvin_tag> from_celcius(
```

Comment:
That occurs because this returns a `quantity<int, temperature_in_kelvin_tag>`, so it is truncated.
Perhaps the correct thing to do is return a `quantity<units::temperature::value_type, temperature_in_kelvin_tag>`, which means all these functions will return `quantity<float, temperature_in_kelvin_tag>` (or whatever is used, but currently float).